modules/reference/pages/data-transform-rust-sdk.adoc

```diff
@@ -0,0 +1,134 @@
+= Rust SDK for Data Transforms
+:description: Work with data transforms using Rust.
+
+Rust SDK reference documentation for Redpanda data transforms.
+
+The `redpanda_transform_sdk` crate contains the core functions and types for transforming data within Redpanda.
+
+== Functions
+
+The Rust SDK includes the following https://doc.rust-lang.org/rust-by-example/fn.html[function] to manage data transforms to an output topic.  
+
+=== on_record_written
+
+[,rust]
+----
+pub fn on_record_written<E, F>(cb: F) -> !
+where
+    E: Debug,
+    F: Fn(WriteEvent<'_>, &mut dyn RecordWriter) -> Result<(), E>,
+----
+
+The `on_record_written()` function transforms individual records after they have been written to an input topic. The function registers a callback to be fired when a record is written to the topic. This callback is triggered after the record has been written and fsynced to disk and the producer has been acknowledged (see <<writeevent,`WriteEvent`>>).
+
+Since this method blocks and runs forever, it should be called from the `main()` function. Perform the necessary steps before transforming any records before calling this method.
```

Comment:
Not sure what this sentence should say but seems like a typo:

> Perform the necessary steps **before** transforming any records **before** calling this method.

Reply:
I'd prefer to start this paragraph with the action required followed by the reason:

```
NOTE: You must call this method from the `main()` function. REASON...
```